docs/DeveloperGuide.md

```diff
@@ -4,26 +4,162 @@
+6. The information retrieved will then be formatted and returned to screen for the user the see the information.
+
+### Add Goods Method
+#### Description
+Add Goods belongs as part of the Commands Class. It is used to add a Good Object into the Collection of Goods Objects currently in the Warehouse.
+
+#### Operation
+![Regex Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/addGood.puml)
+The above sequence diagram shows the operation of how the add goods method will be called. 
+1. The User input will be read by the User Interface Class
+2. The User Interface Class will then match the command keyword `add`
+3. The Regex Class will then be call to match the rest of the user's input to find the necessary values required for the Good Class
+4. Afterwards, the User Interface class will call addGoods() method from the Commands Class
+5. This method will then call the constructor of the Good class, creating a new instance of the Good Object
+6. The addGood() method will then add the Good to the Collections storing the Good Objects currently in the WareHouse
+
+For more examples of how a user can use a command, refer to the [UserGuide](/UserGuide.md)
+
+### Remove Goods Method
+#### Description
+Remove Goods belongs as part of the Commands Class. It is used to remove a certain amount of goods from the inventory.
+#### Operation
+![removeGood diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/removeGood.puml)
+The above sequence diagram shows the operation of how the add goods method will be called.
+1. The User input will be read by the User Interface Class
+2. The User Interface Class will then match the command keyword `remove`
+3. The Regex Class will then be called to match the rest of the user's input to find the values required to remove goods from the inventory.
+4. Afterwards, the User Interface class will call removeGood() method from the Commands Class
+5. This method will then reduce the quantity of a type of goods if the quantity input is not larger than the existing quantity. If the quantity input is the same as the existing quantity, the goods object will be removed from the inventory.
+6. The UI will show a message of format
+    > `quantity` `name` have been removed
+to show that the operation is successful.
+
+For more examples of how a user can use a command, refer to the [UserGuide](/UserGuide.md)
+
+### List Goods or List Orders Method
+#### Description
+The list goods or list orders method belongs to the Warehouse Class. It is used to display the list of existing goods 
+or existing orders in the warehouse to the User.
+
+![List sequence Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/List.puml)
+
```

Comment:
The activation bar for the first condition should not spill over into the "else" condition

![image](https://user-images.githubusercontent.com/79437795/160970631-8623768c-b229-4a83-a229-f2d1628190a4.png)